### PR TITLE
@kanaabe Add Sailthru meta tags on the show page

### PR DESCRIPTION
**Pull request by starsirius**
Closes https://github.com/artsy/force/issues/124
Metaphysics dependency: https://github.com/artsy/metaphysics/pull/411

Spoke with Sara and decided to change definition of some tags. I will add more details inline. Thanks for helping with the integration.

### CODE REVIEW

File: apps/show/helpers/view_helpers.coffee [hunk 1]
```diff
@@ -239,6 +240,12 @@ module.exports =
       ]).join(' ')
     ]).join(', ')
 
+  fairOrShowLocation: (show) ->
+    if show.fair?
+      show.fair.location
+    else
+      show.location
```
Comment (starsirius) [inline apps/show/helpers/view_helpers.coffee]: When the show is a partner show, we want it's location. When it's a fair booth, we want the the location of the fair.

File: apps/show/helpers/view_helpers.coffee [hunk 2]
```diff
@@ -269,3 +276,19 @@ module.exports =
       artistText
       info
     ]).join(' ')
+
+  bestAddress: (location) ->
+    location.display or @displayAddress(location)
```
Comment (starsirius) [inline apps/show/helpers/view_helpers.coffee]: The location returned from Gravity isn't really consistent across models. Let's wrap it in a helper here where we can keep tweaking, if necessary.

File: apps/show/routes.coffee [hunk 3]
```diff
@@ -18,6 +18,10 @@ query = """
       description
       status
       href
+      counts {
+        artworks
+        eligible_artworks
```
Comment (starsirius) [inline apps/show/routes.coffee]: Grabbing from https://github.com/artsy/metaphysics/pull/411.

File: apps/show/templates/meta.jade [hunk 4]
```diff
@@ -1,24 +1,51 @@
...
-meta( property="og:title", content= title )
 meta( name="description", content= description )
-meta( property="og:description", content= description )
-meta( property="twitter:description", content= description )
 link( rel="canonical", href="#{sd.APP_URL}#{partner_show.href}" )
 link( rel="alternate", media= sd.MOBILE_MEDIA_QUERY, href="#{sd.CANONICAL_MOBILE_URL}#{partner_show.href}" )
-meta( property="twitter:card", content="summary" )
+
+//- Open Graph
+meta( property="og:title", content= title )
```
Comment (starsirius) [inline apps/show/templates/meta.jade]: Reorganized meta tags based on purposes.

File: apps/show/templates/meta.jade [hunk 5]
```diff
@@ -1,24 +1,51 @@
...
+meta( name="sailthru.show.type", content= ViewHelpers.sailthruShowType(partner_show) )
+meta( name="sailthru.title", content= title )
+meta( name="sailthru.description", content= description )
+meta( name="sailthru.date", content= moment(partner_show.start_at).toISOString() )
+meta( name="sailthru.expire.date", content= moment(partner_show.end_at).add(1, 'day').toISOString() )
+meta( name="sailthru.tags", content= ViewHelpers.sailthruTags(partner_show).join(", ") )
+if partner_show.fair && partner_show.location
+  meta( name="sailthru.fair.location", content= partner_show.location.display )
+if partner_show.counts
+  meta( name="sailthru.show.works.number", content= partner_show.counts.eligible_artworks )
```
Comment (starsirius) [inline apps/show/templates/meta.jade]: Only published artworks.

File: apps/show/test/helpers/view_helpers.coffee [hunk 6]
```diff
@@ -0,0 +1,22 @@
+ViewHelpers = require '../../helpers/view_helpers'
+{ fabricate } = require 'antigravity'
+
+describe 'ViewHelpers', ->
```
Comment (starsirius) [inline apps/show/test/helpers/view_helpers.coffee]: Add some minimum tests for the helpers.

**Comment by kanaabe** [comment]
Nice usage of `ViewHelpers`, looks 💯

**Comment by garrengotthardt** [comment]
[Adding Sailthru metatags on show pages to facilitate email integration](https://trello.com/c/PMsTJGJE/581-adding-sailthru-metatags-on-show-pages-to-facilitate-email-integration)

**Comment by sperle** [comment]
Would it be possible to change "date" to the actual publish date (the date it's live on the website) rather than the date the show opens? we've come across an issue, particularly with promoted content but all shows, where we're promoting shows before they're open but are live on the site. It's a big problem with tomorrow's miami email for gagosian and another promoted show. We can handle manually if necessary but if it were a quick fix it would be a HUGE help.  

cc @starsirius

**Comment by starsirius** [comment]
@sperle I am going to work on this today. It requires changes on Metaphysics but should be fine. Will cc you in the PR.

**Comment by starsirius** [comment]
@sperle It turned out to be a bigger issue. The [`publish_at`](https://github.com/artsy/gravity/blob/bfb8207afbdc2383d4b8ff0bbb884ef2f6a3c877/app/models/domain/partner_show.rb#L21) timestamp was historically used in the Eigen featured shows feed (I am not sure if that's still used, cc @alloy). A large portion of the shows don't have that set.

From what I understand Force only uses the [`displayable`](https://github.com/artsy/metaphysics/blob/e9c34c4879add044929ce5669569eeb2215d9bbe/schema/partner_show.js#L272) boolean field to determine whether to display a show or not, which means we don't have a timestamp for when it's initially set to `displayable: true`. Is including that boolean field for Sailthru useful?

**Comment by alloy** [comment]
> The publish_at timestamp was historically used in the Eigen featured shows feed (I am not sure if that's still used, cc @alloy).

A quick search does not yield any occurrences in Eigen’s codebase.

**Comment by sperle** [comment]
@starsirius not 100% sure I follow but the only relevant field for this matter is 'date' -- if you passed the actual open date as a custom var the way most of the other fields are passed and just literally made 'date' whatever date it is at the time this is happening, that would be sufficient for my purposes.

**Comment by starsirius** [comment]
- [ ] Change `sailthru.date` to now.
- [ ] Change `sailthru.expire.date` to `sailthru.expire_date`.
- [ ] Change title to just show title instead of page title.
- [ ] Add `sailthru.open.at` and `sailthru.end.at`.

**Pull request merged**